client/blocks/announcement-modal/docs/example.jsx

```diff
@@ -1,20 +1,56 @@
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PlaceholderImage from 'calypso/assets/images/marketplace/plugins-revamp.png';
 import AnnouncementModal from 'calypso/blocks/announcement-modal';
-import PlaceholderImage from 'calypso/images/extensions/woocommerce/image-placeholder.png';
+import Button from 'calypso/components/forms/form-button';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference, hasReceivedRemotePreferences } from 'calypso/state/preferences/selectors';
+
+const AnnouncementModalExample = () => {
+	const dispatch = useDispatch();
+	const userId = useSelector( ( state ) => getCurrentUserId( state ) );
+	const hasPreferences = useSelector( hasReceivedRemotePreferences );
+	const announcementId = 'example';
+	const dismissPreference = `announcement-modal-${ announcementId }-${ userId }`;
+	const isDismissed = useSelector( ( state ) => getPreference( state, dismissPreference ) );
+	const [ show, setShow ] = useState( false );
+
+	useEffect( () => {
+		if ( ! hasPreferences ) {
+			return;
+		}
+
+		if ( isDismissed ) {
+			// Override the dismissing mechanism of the AnnouncementModal and just utilize useState here.
+			dispatch( savePreference( dismissPreference, 0 ) );
+			setShow( false );
```

Comment:
This still works a bit off - try the following steps quickly:

* Click "Show".
* Click "Close".
* Click "Show" again.
* Wait a sec. The modal will close, although you haven't clicked on "Close".

Reply:
The `savePreference` inside the component
https://github.com/Automattic/wp-calypso/blob/0178a4ae1aa2031071dc846cfc0344b353fd233e/client/blocks/announcement-modal/index.jsx#L51

saves the value locally and triggers a network request.
- replaced `savePreference` with `setPreference` in the example so that we don't trigger unnecessary network requests
- removed any logic that would rerender the example component
- used a setTimeout to make sure the `setPreference( dismissPreference, 0 )` event is called last